app.py

```diff
@@ -15,9 +15,114 @@
 # You should have received a copy of the GNU General Public License
 # along with this program. If not, see <http://www.gnu.org/licenses/>.
 
-"""This is the main script of the template project."""
+"""Graph metrics exporter logic for the Thoth project."""
+
+import os
+import logging
+
+import click
+from enum import Enum
+
+from prometheus_client import CollectorRegistry, Gauge, push_to_gateway
+
+from thoth.common import init_logging
+from thoth.storages import GraphDatabase
+from thoth.storages import __version__ as __storages_version__
+from thoth.common import __version__ as __common_version__
+
+
+__version__ = "0.0.1"
+__service_version__ = f"{__version__}+common.{__common_version__}.storages.{__storages_version__}"
+
+_LOGGER = logging.getLogger("thoth.graph_metrics_exporter")
+
+COMPONENT_NAME = "graph-metrics-exporter"
+
+# metrics
+PROMETHEUS_REGISTRY = CollectorRegistry()
+THOTH_METRICS_PUSHGATEWAY_URL = os.environ["PROMETHEUS_PUSHGATEWAY_URL"]
+THOTH_DEPLOYMENT_NAME = os.environ["THOTH_DEPLOYMENT_NAME"]
+
+
+class TaskEnum(Enum):
+    """Class for the task to be run."""
+
+    CORRUPTION_CHECK = "graph_corruption_check"
+
+init_logging()
+
+database_schema_revision_script = Gauge(
+    "thoth_database_schema_revision_script",
+    "Thoth database schema revision from script",
+    ["component", "revision", "env"],
+    registry=PROMETHEUS_REGISTRY,
+)
+
+graphdb_is_corrupted = Gauge(
+    "thoth_graphdb_is_corrupted",
+    "amcheck has detected corruption.",
+    [],
+    registry=PROMETHEUS_REGISTRY,
+)
+
+
+def _create_common_metrics():
+    """Create common metrics to pushgateway."""
+    database_schema_revision_script.labels(
+        COMPONENT_NAME, GraphDatabase().get_script_alembic_version_head(), THOTH_DEPLOYMENT_NAME
+    ).inc()
+
+
+def _send_metrics():
+    """Send metrics to pushgateway."""
+    try:
+        _LOGGER.debug(f"Submitting metrics to Prometheus pushgateway {THOTH_METRICS_PUSHGATEWAY_URL}")
+        push_to_gateway(
+            THOTH_METRICS_PUSHGATEWAY_URL,
+            job=COMPONENT_NAME,
+            registry=PROMETHEUS_REGISTRY,
+        )
+    except Exception as e:
+        _LOGGER.exception(f"An error occurred pushing the metrics: {str(e)}")
+
+
+def _graph_corruption_check(graph: GraphDatabase):
+    if graph.is_database_corrupted():
+        _LOGGER.info("Graph database is corrupted!")
+        graphdb_is_corrupted.set(1)
+    else:
+        _LOGGER.info("Graph database is not corrupted.")
+        graphdb_is_corrupted.set(0)
+
+
+@click.command()
+@click.option(
+    "--task",
+    "-t",
+    type=str,
+    required=True,
+    help="""Long running task to be performed against database.
+            Current tasks available are:
+            """
+    + ",\n".join([entity.value for entity in TaskEnum])
```

Comment:
You can use `click.Choice`. In such a case, click will automatically validate what a user passed in and that it corresponds to desired task to be run in this case: https://click.palletsprojects.com/en/8.0.x/options/#choice-options

Reply:
Nice!! Thanks 🍺